testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/AbstractJSConsoleExampleAdapterTest.java

```diff
@@ -351,6 +353,42 @@ public void loginRequiredAction() {
         waitUntilElement(jsConsoleTestAppPage.getOutputElement()).text().contains("Init Success (Authenticated)");
     }
 
+    @Test
+    public void testUpdateToken() {
+        logInAndInit("standard");
+
+        jsConsoleTestAppPage.setTimeSkewOffset(-33);
+        setTimeOffset(33);
```

Comment:
Are you sure it is wrong? The point is, I can't increase min validity or reduce the token expiration because I need to move in time forward. With moving forward in time, I am creating the gap between real token expiration and false (KEYCLOAK-3586) token expiration. If the gap would be small, there is a high probability they won't catch KEYCLOAK-3586 issue, because of slow machines etc.